### DISCUSSION TITLE
How do we know if we are respecting the Apache License 2.0 license?

### DISCUSSION THREAD

**Discussion by alexe100** [emqx/emqx#13793] [General]
Hello,
According to EMQX Github page EMQX uses, by default, the Apache License 2.0, see APL.txt. However there are folders where Business Source License applies. Mainly bridges, emqx_dashboard_sso, emqx_dashboard_rback, etc. are governed by Business Source License.
Hence, some feature could be activated (using dashboard or CLI) and then the Business Source License applies.

I we want to comply with Apache License 2.0 license, how do we know if any feature is running uses Business Source License applies? How can we protect ourselves about this?

By the way, what is the cost of the commercial license?

Thanks.

**Comment by id** [comment]
@alexe100 code covered by BSL is only included in the Enterprise packages. If you use Community edition packages, you don't need to worry. If you build EMQX yourself, you need to set PROFILE=emqx environment variable.

**Comment by id** [comment]
About the commercial license, please contact our Sales team https://www.emqx.com/en/contact

**Comment by ieQu1** [comment]
Hello,

Indeed, source code of both open-source and enterprise editions is stored in the same monorepo. However, the build system ensures that only applications covered by Apache license are compiled in the default profile. So open-source edition release doesn't contain any executable binaries derived from the code covered by BSL. 
Therefore, it's impossible to accidentally turn open-source edition into enterprise edition via dashboard or CLI.
